src/mon/MDSMonitor.cc

```diff
@@ -1841,7 +1841,6 @@ void MDSMonitor::update_metadata(mds_gid_t gid,
   bufferlist bl;
   encode(pending_metadata, bl);
   t->put(MDS_METADATA_PREFIX, "last_metadata", bl);
-  paxos.trigger_propose();
```

Comment:
Sounds reasonable. Should we introduce an additional check when a paxos proposal is made, `last_metadata` key should be available (by setting a flag in this routine). The debug introduced in this change is probably just enough to be able to deduce the reasoning.

Reply:
> Should we introduce an additional check when a paxos proposal is made, last_metadata key should be available (by setting a flag in this routine).

Sorry, I don't understand what you mean. Can you elaborate?

Reply:
Sorry for late reply. Since this fix is a hunch, I was proposing if the theory can be validated by adding checks when doing a paxos proposal by checking if `last_metadata` was set but is empty during paxos proposal.

Reply:
I'm not sure how `last_metadata` would get unset at all. The working theory here was that the pending_metadata would get changed but the PAXOS commit would fail. This would cause the in-memory state to mismatch with the committed state. I don't know how we'd reasonably check for that without loading the metadata with each commit, which is probably too expensive.

Reply:
Fair enough.